ssl/t1_lib.c

```diff
@@ -2291,6 +2308,24 @@ int tls_choose_sigalg(SSL *s, int fatalerrs)
         int curve = -1, skip_ec = 0;
 #endif
 
+        /*
+         * Look for a certificate indicated by signature_algorithms_cert,
+         * if present.
+         */
+        if (s->s3->tmp.peer_cert_sigalgs != NULL) {
+            for (i = 0; i < s->s3->tmp.peer_cert_sigalgslen; i++) {
+                lu = tls1_lookup_sigalg(s->s3->tmp.peer_cert_sigalgs[i]);
+                if (lu == NULL || !ssl_has_cert(s, lu->sig_idx))
+                    continue;
+                s->s3->tmp.cert = &s->cert->pkeys[sig_idx];
```

Comment:
Do you mean `lu->sig_idx` here?

Reply:
Yes...and that probably explains the issue with the tests.

Reply:
(actually it doesn't impact the tests, since we don't generate the extension yet)